client/src/App.tsx

```diff
@@ -8,9 +8,13 @@ import {
   Transactions,
   Budget,
   ErrorPage,
+  BudgetMain,
 } from "./pages";
 import "./App.css";
 import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
+import { QueryClientProvider, QueryClient } from "react-query";
+
+const queryClient = new QueryClient();
```

Comment:
So this is React Query - It helps to clean up my code a bit instead of using useState and useEffect, I can destructure the fetch requests more easily and gives me access to hooks that are written specifically for data fetching.

Apparently ReactQuery is also a great way to make caching easier. I don't know much about caching, but I'll be learning more about this